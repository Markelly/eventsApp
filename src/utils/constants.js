export const IP_URL = "https://extreme-ip-lookup.com/json/";
export const DISCOVER_URL = "https://app.ticketmaster.com/discovery/v2";
export const API_KEY = "4diZ7tJQArO9sxHiVWMASl5udVTAoI78";

export const EVENTS_PATH = "/events";
export const FAVORITES_PATH = "/favorites";
export const LOGIN_PATH = "/login";
export const REGISTER_PATH = "/register";

export const SORT_OPTIONS = [
  { text: 'Sort by', value: null, disabled: true },
  { text: 'Upcoming events', value: 'date,asc', disabled: false },
  { text: 'Name', value: 'name,asc', disabled: false },
  { text: 'Relevance', value: 'relevance,asc', disabled: false },
  { text: 'Distance', value: 'distance,asc', disabled: false }
];

export const monthName = [
  "January",
  "February",
  "March",
  "April",
  "May",
  "June",
  "July",
  "August",
  "September",
  "October",
  "November",
  "December"
];

export const weekDayName = [
  "Sunday",
  "Monday",
  "Tuesday",
  "Wednesday",
  "Thursday",
  "Friday",
  "Saturday"
];
