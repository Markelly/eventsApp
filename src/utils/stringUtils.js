import { EVENTS_PATH } from "./constants";

export const getEventsList = eventsList => {
  let event = {};
  let events = [];
  eventsList.forEach(item => {
    event.name = item.name;
    event.date = getDate(item);
    event.price = getPrice(item);
    events.push(event);
    event = {};
  });
  return events;
};

export const getPrice = item => {
  const min = item.priceRanges ? item.priceRanges[0].min : 0;
  const max = item.priceRanges ? item.priceRanges[0].max : 0;
  return item.priceRanges ? `€${min} - €${max}` : "Free";
};

export const getDate = item => {
  let formattedDate = "No date informed";
  const date = new Date(
    item.dates && item.dates.start && item.dates.start.localDate
  );
  if (date) {
    let weekday = getWeekDayName(date.getDay());
    let month = getMonthName(date.getMonth());
    let day = date.getDate();
    let year = date.getFullYear();
    formattedDate = `${weekday}, ${day} ${month} ${year}`;
  }
  return formattedDate;
};

export const getWeekDayName = day => {
  const days = [
    "Sunday",
    "Monday",
    "Tuesday",
    "Wednesday",
    "Thursday",
    "Friday",
    "Saturday"
  ];
  return days[day];
};

export const getMonthName = month => {
  const months = [
    "January",
    "February",
    "March",
    "April",
    "May",
    "June",
    "July",
    "August",
    "September",
    "October",
    "November",
    "December"
  ];
  return months[month];
};

export const isEventsPage = path => {
  return path === EVENTS_PATH;
};

export default {
  getEventsList,
  getDate,
  getPrice,
  getWeekDayName,
  getMonthName,
  isEventsPage
};
