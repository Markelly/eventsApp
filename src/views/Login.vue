<template>
  <div class="login">
    <Form
      title="Login"
      footer-label="Don't have an account? Create"
      footer-link="/register"
      button-label="Login"
      @submit="login"
    >
      <div slot="form-fields">
        <b-form-input v-model="email" type="email" placeholder="Email" />
        <b-form-input
          v-model="password"
          type="password"
          placeholder="Password"
        />
        <Error v-if="error" />
      </div>
    </Form>
  </div>
</template>

<script>
import { mapActions, mapGetters } from "vuex";
import { EVENTS_PATH } from "@/utils/constants";
import Form from "@/layouts/Form";
import Error from "@/components/Error";

export default {
  name: "Login",
  components: { Error, Form },
  data: () => ({
    email: "",
    password: ""
  }),
  created() {
    this.error && this.clearError();
    this.isAuthenticated && this.$router.push(EVENTS_PATH);
  },
  computed: {
    ...mapGetters({
      isAuthenticated: "isAuthenticated",
      error: "error"
    })
  },
  methods: {
    ...mapActions({
      authLogin: "login",
      clearError: "clearAuthError"
    }),
    async login() {
      await this.authLogin({ email: this.email, password: this.password });
      if (this.isAuthenticated && !this.error) {
        this.$router.push(EVENTS_PATH);
      }
    }
  }
};
</script>

<style scoped></style>
