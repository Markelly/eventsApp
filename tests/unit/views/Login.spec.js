import { createLocalVue, shallowMount } from "@vue/test-utils";
import Router from "vue-router";
import { BootstrapVue } from "bootstrap-vue";
import Vuex from "vuex";
import Login from "@/views/Login";
import { EVENTS_PATH } from "@/utils/constants";

const localVue = createLocalVue();
localVue.use(BootstrapVue);
localVue.use(Vuex);
localVue.use(Router);

describe("<login />", () => {
  const user = { email: "user@app.com", password: "123" };
  let wrapper, store, mockRoute, spyDispatch, spyRoutePush;

  beforeEach(() => {
    mockRoute = new Router({});
    store = new Vuex.Store({
      modules: {
        namespaced: true,
        actions: {
          login: jest.fn(),
          clearAuthError: jest.fn()
        },
        state: {
          isAuthenticated: false,
          error: null
        }
      }
    });
    spyRoutePush = jest.spyOn(mockRoute, "push");
    spyDispatch = jest.spyOn(store, "dispatch");
    wrapper = shallowMount(Login, {
      localVue,
      store,
      router: mockRoute,
      data: () => ({ email: user.email, password: user.password })
    });
  });

  test("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should redirect to events page if user logged", () => {
    store.getters.isAuthenticated = () => true;
    wrapper = shallowMount(Login, {
      localVue,
      store,
      router: mockRoute
    });
    expect(spyRoutePush).toHaveBeenCalledWith(EVENTS_PATH);
  });

  test("should dispatch an action to login", () => {
    wrapper.vm.login();
    expect(spyDispatch).toHaveBeenCalledWith("login", user);
  });
});
