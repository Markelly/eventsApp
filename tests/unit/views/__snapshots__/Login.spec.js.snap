// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<login /> should render the component 1`] = `
<div class="login">
  <form-stub title="Login" buttonlabel="Login" footerlabel="Don't have an account? Create" footerlink="/register">
    <div>
      <b-form-input-stub value="user@app.com" placeholder="Email" debounce="0" type="email"></b-form-input-stub>
      <b-form-input-stub value="123" placeholder="Password" debounce="0" type="password"></b-form-input-stub>
      <!---->
    </div>
  </form-stub>
</div>
`;
